### ISSUE TITLE
Debian mirrors not synced correctly

### ISSUE THREAD

**Issue by Derek000** [dtag-dev-sec/tpotce#358]
# Issues
I have been attempting to get T-Pot going on Proxmox 5.4-3 using the recently released .iso and on an existing Debian (Stretch and Buster) OS.
It appears that during the install process the OS is upgraded to Buster, which isn't supported by the installer. Therefore the install fails.

As the installer failed during the .ISO init process docker, curl and a range of other applications failed to install.
When attempting to install them the OS update occurred bring it to Buster. After adding the missing software required for the install, the install script then stopped working as the OS was now Buster.

Is there any proven method to get it going on Proxmox?
The method shown on YouTube doesn't appear to work any longer.

I really want to use t-pot.

Thanks 
Derek
<br>
<br>
<br>

<a name="info"></a>
## ⚠️ Basic support information (commands are expected to run as `root`)

- What version of the OS are you currently using `lsb_release -a` and `uname -a`?
Distributor ID: Debian
Description:    Debian GNU/Linux 10 (buster)
Release:        10
Codename:       buster

- What T-Pot version are you currently using?
T-Pot 19.03

- What edition (Standard, Nextgen, etc.) of T-Pot are you running?
Standard

- What architecture are you running on (i.e. hardware, cloud, VM, etc.)?
Proxmox 
500G SSD
8 core i7
32G Memory

- Did you have any problems during the install? If yes, please attach `/install.log` `/install.err`.
[install.log](https://github.com/dtag-dev-sec/tpotce/files/3109960/install.log)

[install.err.log](https://github.com/dtag-dev-sec/tpotce/files/3109964/install.err.log)

- How long has your installation been running?
20 min

- Did you install upgrades, packages or use the update script?
I attempted to do both.

- Did you modify any scripts or configs? If yes, please attach the changes.
No

- Please provide a screenshot of `glances` and `htop`.
![top](https://user-images.githubusercontent.com/16163075/56622668-326ec000-6670-11e9-88fb-871be7e14bda.JPG)

- How much free disk space is available (`df -h`)?
 df -h
Filesystem      Size  Used Avail Use% Mounted on
udev            3.9G     0  3.9G   0% /dev
tmpfs           789M   78M  711M  10% /run
/dev/sda2        71G  5.9G   62G   9% /
tmpfs           3.9G     0  3.9G   0% /dev/shm
tmpfs           5.0M     0  5.0M   0% /run/lock
tmpfs           3.9G     0  3.9G   0% /sys/fs/cgroup
tmpfs           789M     0  789M   0% /run/user/1000

- What is the current container status (`dps.sh`)?
dps.sh
========| System |========
    Date:  Tue 23 Apr 2019 11:37:17 PM UTC
  Uptime:  23:37:17 up 11:39,  2 users,  load average: 0.08, 0.16, 0.17

NAME                  STATUS                       PORTS
adbhoney              DOWN
ciscoasa              DOWN
conpot_guardian_ast   DOWN
conpot_iec104         DOWN
conpot_ipmi           DOWN
conpot_kamstrup_382   DOWN
cowrie                DOWN
cyberchef             DOWN
dionaea               DOWN
elasticpot            DOWN
elasticsearch         DOWN
ewsposter             DOWN
head                  DOWN
heralding             DOWN
honeytrap             DOWN
kibana                DOWN
logstash              DOWN
mailoney              DOWN
medpot                DOWN
nginx                 DOWN
p0f                   DOWN
rdpy                  DOWN
snare                 DOWN
spiderfoot            DOWN
suricata              DOWN
tanner                DOWN
tanner_api            DOWN
tanner_phpox          DOWN
tanner_redis          DOWN
tanner_web            DOWN

- What is the status of the T-Pot service (`systemctl status tpot`)?
systemctl status tpot
● tpot.service - tpot
   Loaded: loaded (/etc/systemd/system/tpot.service; enabled; vendor preset: enabled)
   Active: activating (auto-restart) (Result: exit-code) since Tue 2019-04-23 23:37:44 UTC; 2s ago
  Process: 15568 ExecStartPre=/opt/tpot/bin/updateip.sh (code=exited, status=0/SUCCESS)
  Process: 15587 ExecStartPre=/bin/bash -c /opt/tpot/bin/clean.sh on (code=exited, status=0/SUCCESS)
  Process: 15632 ExecStartPre=/usr/bin/docker-compose -f /opt/tpot/etc/tpot.yml down -v (code=exited, status=0/SUCCESS)
  Process: 15635 ExecStartPre=/usr/bin/docker-compose -f /opt/tpot/etc/tpot.yml rm -v (code=exited, status=0/SUCCESS)
  Process: 15638 ExecStartPre=/bin/bash -c docker network rm $(docker network ls -q) (code=exited, status=1/FAILURE)
  Process: 15658 ExecStartPre=/bin/bash -c docker volume rm $(docker volume ls -q) (code=exited, status=1/FAILURE)
  Process: 15674 ExecStartPre=/bin/bash -c docker rm -v $(docker ps -aq) (code=exited, status=1/FAILURE)
  Process: 15689 ExecStartPre=/bin/bash -c docker rmi $(docker images | grep "<none>" | awk '{print $3}') (code=exited, status=1/FAILU
  Process: 15707 ExecStartPre=/bin/bash -c /sbin/ethtool --offload $(/sbin/ip address | grep "^2: " | awk '{ print $2 }' | tr -d [:pun
  Process: 15714 ExecStartPre=/bin/bash -c /sbin/ethtool -K $(/sbin/ip address | grep "^2: " | awk '{ print $2 }' | tr -d [:punct:]) g
  Process: 15720 ExecStopPost=/opt/tpot/bin/rules.sh /opt/tpot/etc/tpot.yml unset (code=exited, status=0/SUCCESS)

- What ports are being occupied? Stop T-Pot `systemctl stop tpot` and run `netstat -tulpen`
netstat -tulpen
Active Internet connections (only servers)
Proto Recv-Q Send-Q Local Address           Foreign Address         State       User       Inode      PID/Program name
tcp        0      0 0.0.0.0:64295           0.0.0.0:*               LISTEN      0          14368      405/sshd
tcp        0      0 0.0.0.0:61209           0.0.0.0:*               LISTEN      0          13200      389/python3
tcp6       0      0 :::64295                :::*                    LISTEN      0          14370      405/sshd
udp        0      0 0.0.0.0:68              0.0.0.0:*                           0          12909      368/dhclient

- If a single container shows as `DOWN` you can run `docker logs <container-name>` for the latest log entries

**Comment by TheHADILP** [comment]
Hi Derek,
I tried it this morning on my Proxmox System and it doesn't appear to be an issue with Proxmox.
I used the tpot.iso from GitHub and created a VM using that iso file.
It took some time but finally it sucessfully completed the installation and everything was up and running.

Looking at your install.err.log there seems to be a problem with your package mirror (http://mirror.amaze.com.au).
Maybe you can try selecting another package mirror during installation?

Let me know if this fixes the issue.

**Comment by Derek000** [comment]
I’ll give it another go. Thanks for verifying it for me.

**Comment by Derek000** [comment]
It’s interesting, on the same hardware and VM products I used last week, i am now receiving an error that the repositories do not have the necessary Kernel modules for the T-Pot prebuilt ISO kernel version.

**Comment by t3chn0m4g3** [comment]
Some of the Debian mirrors seem to be synced incorrectly. I disabled the automatic mirror selection to adress this Debian related issue. Also Debian released a new mini.iso (which T-Pot's ISO installer is based on) along with other updates in the stable branch.
I swapped the ISO in the release for 19.03. Just download the release ISO and you are good to go.

**Comment by t3chn0m4g3** [comment]
Fixed in 9ec1992929c3b615dfea7d5bbf711d4f4021f342

**Comment by t3chn0m4g3** [comment]
Assuming this is resolved, however @Derek000 if this still does not work for you I recommend you install the latest Debian 9.x stable, clone the tpotce repo and install T-Pot on top.

**Comment by Derek000** [comment]
Hi @t3chn0m4g3 

Yes this is resolved. It looks like the original ISO I downloaded was corrupted in some way.
Using the same DL server and Virtual Hardware/install at my end, I deleted the original VM and installed a new one with a newly downloaded ISO of the same version and it just worked.

I had previously installed t-pot about 10 times with no luck.
Maybe the installer could have some sort of md5 checker on it, if the ISO can be damaged during download.

Thanks again. 
Derek
